### ISSUE TITLE
Cross-compile calico for arm64 and ppc64le from amd64

### ISSUE THREAD

**Issue by deitch** [projectcalico/go-build#24]
Following the merge of #23 , which simplifies building go-build for arm64 on arm64 and ppc64le on ppc64le, this issue is intended as a tracking issue for cross-compile binaries and cross-build of docker images for arm64 and ppc64le (in addition to amd64, of course) from amd64.

Since every CI-as-a-service provides amd64 only, we want to be able to run the entire build and push from amd64. 

Related: see https://github.com/projectcalico/go-build/issues/22

cc @caseydavenport @tomdee @fasaxc 

Ideas and issues raised by the above people are in a follow-on comment.

**Comment by deitch** [comment]
There are two ways to compile from amd64 to arm64:

* `GOARCH=aarch64 go build ...`
* run in a container with `qemu` providing an emulated architecture. See https://blog.hypriot.com/post/setup-simple-ci-pipeline-for-arm-images/ as posted by @liljenstolpe and @ahrkrak 

The first option generally is preferred, but will not always work. Notably, anything with `CGO` may be an issue.
Additionally, if we need to install additional packages using `apk`, we may have issues.

Note that `apk` does support copying files into alternate locations for alternate arch, using `--arch <arch> --root <root>`, e.g. `apk --arch aarch64 --root /var/alt/pkg/aarch64/ add <package>`. This may provide a viable option, especially with multi-stage container builds.

An example of emu-based building is from flannel, notably [here](https://github.com/coreos/flannel/blob/ff1b89d755d5fe41add70ad6e5e75f1a9aae9de5/Makefile#L48)

**Comment by djlwilder** [comment]
I don't see the need  to install packages for the non-native arch into the container.  Everything executed should be amd64 amd64 with the exception of binaries we just built and sometimes we need to do that  therefore qemu is needed in the go-build:amd64 container.  

I played with this approach and will push a RFC to go-build to add qemu-statuc-user.  Using this update I was able to build a ppc64le felix container on amd64.  It takes a bunch of other changes to felix to as well make this work  but its a start.
